openedx/api.py

```diff
@@ -712,3 +714,35 @@ def unsubscribe_from_edx_course_emails(user, course_run):
     except Exception as exc:
         raise UnknownEdxApiEmailSettingsException(user, course_run, exc) from exc
     return result
+
+
+def username_exists_in_openedx(username):
+    """
+    Returns true if the username exists within Open edX.
+    Returns false if the username does not exists in Open edX.
+
+    Args:
+        username (str): the username
+
+    Raises:
+        EdxApiRegistrationValidationException: Raised if the edX API HTTP request fails
+    """
+
+    req_session = requests.Session()
+    req_session.headers.update(
+        {ACCESS_TOKEN_HEADER_NAME: settings.MITX_ONLINE_REGISTRATION_ACCESS_TOKEN}
+    )
+    resp = req_session.post(
+        edx_url(OPENEDX_VALIDATION_REGISTRATION_PATH),
+        data=dict(
+            username=username,
+        ),
+    )
+    if resp.status_code != status.HTTP_200_OK:
+        raise EdxApiRegistrationValidationException(username, resp)
+    result = resp.json()
+    return result["validation_decisions"][
+        "username"
+    ] == "It looks like {} belongs to an existing account. Try again with a different username.".format(
```

Comment:
In my opinion, we should keep this message the same for the username validation error. There is [an existing message](https://github.com/mitodl/mitxonline/blob/main/users/serializers.py#L98) when the username exists. 

Personally, I like this string better, Maybe we can move this to a constant and use it in both places.

Reply:
This is the string which comes in the response from Edx when there is an existing username found.  So this message here doesn't actually end up going anywhere, it's just used for checking Edx response.